src/NServiceBus.Core/Audit/InvokeAuditPipelineBehavior.cs

```diff
@@ -16,6 +16,8 @@ public InvokeAuditPipelineBehavior(string auditAddress, TimeSpan? timeToBeReceiv
 
         public async Task Invoke(IIncomingPhysicalMessageContext context, Func<IIncomingPhysicalMessageContext, Task> next)
         {
+            context.Extensions.Set(AuditProcessingStatisticsBehavior.ProcessingStartedKey, DateTimeOffset.UtcNow);
```

Comment:
I wished we could get this timestamp from the transport via https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Transports/MessageContext.cs

Reply:
we could add it to the `MainPipelineExecutor` to not put the burden on every transport but have it set as early as possible in the pipeline.

Reply:
I like that and it would align with https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs#L23

Reply:
I think figuring out the right place where to put it depends on the discussion whether the statistics are optional or not.